File: 250330/좌표평면 위의 특정 구역 2/specific-zone-above-the-2d-coordinate-2.js
const fs = require("fs");
const input = fs.readFileSync(0).toString().trim().split('\n');

const n = Number(input.shift());
const points = input.slice(0, n).map(line => line.split(' ').map(Number));
    
let answer = Number.MAX_SAFE_INTEGER

for(let i = 0; i < points.length; i++){
    const newArr = points.filter((_, idx) => idx !== i);
    let minX = Number.MAX_SAFE_INTEGER;
    let minY = Number.MAX_SAFE_INTEGER;
    let maxX = Number.MIN_SAFE_INTEGER;
    let maxY = Number.MIN_SAFE_INTEGER;

    for(let j = 0; j < newArr.length; j++){
        const [x, y] = newArr[j];
        minX = Math.min(minX, x);
        minY = Math.min(minY, y);

        maxX = Math.max(maxX, x);
        maxY = Math.max(maxY, y);
    }
    answer = Math.min(answer, (maxX - minX) * (maxY - minY))
}
console.log(answer)

File: 250810/좌표평면 위의 특정 구역 2/specific-zone-above-the-2d-coordinate-2.py
import sys
n = int(input())
points = [tuple(map(int, input().split())) for _ in range(n)]
x = [p[0] for p in points]
y = [p[1] for p in points]


ans = sys.maxsize
for i in range(n):
    
    min_x, min_y = sys.maxsize, sys.maxsize
    max_x, max_y = 1, 1
    for j, (x, y) in enumerate(points):
        if j == i:
            continue
        min_x, min_y = min(min_x, x), min(min_y, y)
        max_x, max_y = max(max_x, x), max(max_y, y)
    
    ans = min(ans, (max_x - min_x) * (max_y - min_y))

print(ans)
        
